lua/lspconfig/vala_ls.lua

```diff
@@ -5,9 +5,25 @@ configs.vala_ls = {
   default_config = {
     cmd = {'vala-language-server'},
     filetypes = {'vala', 'genie'},
-    root_dir = function(fname)
-      return util.root_pattern("meson.build")(fname)
-        or util.find_git_ancestor(fname)
+    root_dir = function (fname)
+      local root = util.search_ancestors(fname, function (path)
+        local pattern = "meson.build"
+        local f = vim.fn.glob(util.path.join(path, pattern))
+        if not util.path.exists(f) then
```

Comment:
You can skip a syscall by just checking if `f` is an empty string since `glob` has already verified that the path exists.

```suggestion
        if f ~= '' then
```